src/ts/components/subtitleoverlay.ts

```diff
@@ -1,33 +1,92 @@
 import {Container, ContainerConfig} from './container';
-import {UIInstanceManager} from '../uimanager';
+import {UIInstanceManager, UISubtitleConfig} from '../uimanager';
 import SubtitleCueEvent = bitmovin.PlayerAPI.SubtitleCueEvent;
 import {Label, LabelConfig} from './label';
 import {ComponentConfig, Component} from './component';
 import {ControlBar} from './controlbar';
+import {ColorUtils, StorageUtils} from '../utils';
+import {DOM} from '../dom';
+
+export interface SubtitleOverlayConfig extends ContainerConfig {
+  subtitleConfig?: UISubtitleConfig;
+}
 
 /**
  * Overlays the player to display subtitles.
  */
-export class SubtitleOverlay extends Container<ContainerConfig> {
+export class SubtitleOverlay extends Container<SubtitleOverlayConfig> {
+
+  public config: SubtitleOverlayConfig;
+
+  private subtitleManager: ActiveSubtitleManager;
+  private previewSubtitleActive: boolean = false;
+  private previewSubtitle: SubtitleLabel = new SubtitleLabel({text: 'example subtitle'});
 
   private static readonly CLASS_CONTROLBAR_VISIBLE = 'controlbar-visible';
 
-  constructor(config: ContainerConfig = {}) {
+  private fontColor?: ColorUtils.Color;
+  private backgroundColor?: ColorUtils.Color;
+  private windowColor?: ColorUtils.Color;
+  private fontFamily?: string;
+  private fontStyle?: string;
+  private fontVariant?: string;
+  private characterEdge?: string;
+  /**
+   * Font size is defined by a base size
+   * and a multiplicating coefficient depending
+   * on user preferences
+   **/
+  private fontCoefficient?: number;
+  private fontSize?: number;
+
+  constructor(config: SubtitleOverlayConfig = {}) {
     super(config);
 
-    this.config = this.mergeConfig(config, {
-      cssClass: 'ui-subtitle-overlay',
-    }, this.config);
+    this.config = this.mergeConfig( config, {
+        cssClass: 'ui-subtitle-overlay',
+      }, this.config);
+
+    config = this.config;
   }
 
   configure(player: bitmovin.PlayerAPI, uimanager: UIInstanceManager): void {
     super.configure(player, uimanager);
 
+    let config = <SubtitleOverlayConfig>this.config;
+
+    this.fontColor = ColorUtils.foreground;
+    this.backgroundColor = ColorUtils.background;
+    this.windowColor = ColorUtils.background;
+    this.fontFamily = '';
+    this.fontStyle = '';
+    this.fontVariant = 'normal';
+    this.characterEdge = '';
+    this.fontCoefficient = 1;
+    this.fontSize = 1.2;
+
+    // Update the config first loading info from UImanager config
+    // then overwrites it with config given to the component if it applies
+    // finally loads user preferences from local storage
+    this.updateSubtitleOverlayFromUIconfig(uimanager.getConfig().subtitles);
```

Comment:
The global `UIConfig` takes precendence over the local `SubtitleOverlayConfig`, please flip the lines.